src/system/mod.rs

```diff
@@ -266,10 +266,15 @@ pub fn set_target_user(
         cmd.pre_exec(move || {
             cerr(libc::setgroups(
                 target_user.groups.len(),
-                target_user.groups.as_ptr(),
+                target_user
+                    .groups
+                    .iter()
+                    .map(|g| g.get())
+                    .collect::<Vec<_>>()
+                    .as_ptr(),
```

Comment:
This is allocating after fork(), which can be problematic in certain cases as malloc is not async-signal-safe. I think it would be fine in the current code, but still would be nice to avoid it. One way would be the following:

```suggestion
                target_user.groups.as_ptr().cast::<gid_t>()
```

taking advantage of the `#[repr(transparent)]` you added to `GroupId`. If you do this, you will need to disregard my previous comment about `#[repr(transparent)]` and document here that this is fine because of the `#[repr(transparent)]` on `GroupId`. Maybe also add documentation to the `#[repr(transparent)]` on `GroupId` to point to this location as a place that depends on it.